src/main/java/shikabot/parser/Parser.java

```diff
@@ -0,0 +1,142 @@
+package shikabot.parser;
+
+import shikabot.command.AddCommand;
+import shikabot.command.Command;
+import shikabot.command.DeleteCommand;
+import shikabot.command.DoneCommand;
+import shikabot.command.ExitCommand;
+import shikabot.command.FailedCommand;
+import shikabot.command.InvalidCommand;
+import shikabot.command.ListCommand;
+import shikabot.task.Task;
+
+public class Parser {
+
+    public static final int INVALID_DEADLINE_SYNTAX = 1;
+    public static final int INVALID_EVENT_SYNTAX = 2;
+    public static final int NUMBER_FORMAT_ERROR = 3;
+    public static final int NEGATIVE_INDEX_ERROR = 4;
+    public static final int INVALID_TASK = 5;
+    public static final int EMPTY_FIELD = 6;
+
+    private static final int DONE_LENGTH = 4;
+    private static final int TODO_LENGTH = 4;
+    private static final int DEADLINE_LENGTH = 8;
+    private static final int EVENT_LENGTH = 5;
+    private static final int DELETE_LENGTH = 6;
+    private static final int DIVIDER_LENGTH = 3;
+
+    public boolean isAddCommand(String text) {
+        return text.startsWith("todo") || text.startsWith("deadline") || text.startsWith("event");
+    }
+
+    public Command parseCommand(String text) {
+        Command command;
+        text = text.trim();
+        if (text.equals("bye")) {
+            command = new ExitCommand();
+        } else if (text.equals("list")) {
+            command = new ListCommand();
+        } else if (text.startsWith("done")) {
+            command = parseDoneCommand(text);
+        } else if (text.startsWith("delete")) {
+            command = parseDeleteCommand(text);
+        } else if (isAddCommand(text)) {
+            command = parseAddCommand(text);
+        } else {
+            command = new InvalidCommand();
+        }
```

Comment:
perhaps it would look neater if you use switch case instead of ifelse. This is fine too